feat(run): Expose platform field and move machine strategy initialization

LucaSeri opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

Commit:
feat(run): Add public platform field

This commit enables users to manually set the platform when
programatically running an application.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

Commit:
fix(run): Always discover platform machine controller strategy

Attempting to call run programatically would result in a nil pointer
dereference, as the machine controller was only set in Pre, which is
only called when using run from the cli.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

nderjung commented on internal/cli/kraft/run/run.go:
```diff
@@ -51,7 +51,7 @@ type RunOptions struct {
 
 	workdir           string
-	platform          mplatform.Platform
+	Platform          mplatform.Platform `noattribute:"true"`
```
Can you move this up into the other "exported" fields and sort alphabetically? :pray:

nderjung commented:
Hi, just going through this again and it occurs to me that `Architecture` and `Platform` options are now two different types. I think it makes sense to use a single type (`string`) for both. Could you turn the exposed `Platform` non-attribute to a string, and if it is populated seed the private attribute `platform` with the `mplatform.Platform` type?

Commit:
fix(run): Remove noop assignment

This assignment is always overwritten when discovering
the platform machine strategy.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

LucaSeri commented:
@nderjung I added the suggested change.
